lib/excon/socket.rb

```diff
@@ -24,119 +25,44 @@ def initialize(data = {})
       @nonblock = data[:nonblock]
       @read_buffer = ''
       @eof = false
-
       connect
     end
 
-    def read(max_length=nil)
-      if @eof
-        return max_length ? nil : ''
-      elsif @nonblock
-        begin
-          if max_length
-            until @read_buffer.length >= max_length
-              @read_buffer << @socket.read_nonblock(max_length - @read_buffer.length)
-            end
-          else
-            while true
-              @read_buffer << @socket.read_nonblock(@data[:chunk_size])
-            end
-          end
-        rescue OpenSSL::SSL::SSLError => error
-          if error.message == 'read would block'
-            if IO.select([@socket], nil, nil, @data[:read_timeout])
-              retry
-            else
-              raise(Excon::Errors::Timeout.new("read timeout reached"))
-            end
-          else
-            raise(error)
-          end
-        rescue Errno::EAGAIN, Errno::EWOULDBLOCK, IO::WaitReadable
-          if @read_buffer.empty?
-            # if we didn't read anything, try again...
-            if IO.select([@socket], nil, nil, @data[:read_timeout])
-              retry
-            else
-              raise(Excon::Errors::Timeout.new("read timeout reached"))
-            end
-          end
-        rescue EOFError
-          @eof = true
-        end
+    def read(max_length = nil)
+      # if @eof
+      return max_length ? nil : '' if @eof
```

Comment:
It is kind of a minor/preference thing, but could you decompose these into multi-line ifs instead of ternaries? Thanks!